### PR TITLE
Backport help command and command behavior improvements

### PR DESCRIPTION
**Pull request by michaelklishin**
This backports the help command, command list table, related
core infra (CommandBehaviour improvements) and command-specific
documentation from master.

This only changes the help command and --help behavior, which are
considered documentation and thus appropriate for a patch release.

Per discussion with the team.

Plugin-provided CLI commands need to be updated to show up in the new categorized command
list.

### CODE REVIEW

File: lib/rabbitmq/cli/ctl/commands/list_consumers_command.ex [hunk 1]
```diff
@@ -29,7 +29,8 @@ defmodule RabbitMQ.CLI.Ctl.Commands.ListConsumersCommand do
   def info_keys(), do: @info_keys
 
   def merge_defaults([], opts) do
-    {Enum.map(@info_keys, &Atom.to_string/1), Map.merge(%{vhost: "/", table_headers: true}, opts)}
+    {Enum.map(@info_keys -- [:activity_status], &Atom.to_string/1),
```
Comment (hairyhum) [inline lib/rabbitmq/cli/ctl/commands/list_consumers_command.ex]: Should consumer activity status be a part of this PR?
Applies to the rest of the file.

**Comment by michaelklishin** [comment]
Ready for another round.

File: lib/rabbitmq/cli/ctl/commands/list_consumers_command.ex (hunk 1 above)
Comment (lukebakken) [reply lib/rabbitmq/cli/ctl/commands/list_consumers_command.ex]: Good catch!

**Comment by lukebakken** [comment]
> Would it be a good thing to have a test, which will call help for all known commands? To make sure none of them is crashing on help?

@hairyhum that makes sense to me. I suppose it could be separate from this PR, right?

**Comment by hairyhum** [comment]
The default options text was slightly reduced in size, but I agree that it attracts too much attention.

**Comment by lukebakken** [comment]
@michaelklishin @hairyhum see #333 for what I'm thinking.

**Comment by michaelklishin** [comment]
Merging as most of the discussion has shifted to #333 and ideas that do not interfere with this PR.

**Pull request merged**